### ISSUE TITLE
ModuleNotFoundError: No module named 'distutils'

### ISSUE THREAD

**Issue by sagisar1** [fortra/impacket#1856]
Hello, since python 3.12 deprecated distutils, I am not able to use any of impacket tools in my kali linux with python 3.12. I don't know why the python developers did this, but it seems to affect a lot of projects, and I think it's gonna be hard to replace in each project setuptools instead of distutils, but maybe I am wrong, I would like to hear from you if I do. About the issue:

impacket version:  python3-impacket  - 0.12.0-3  , impacket-scripts - 1.10  
Python version:  3.12.7
Target OS:  kali linux 2024.4

**Just to clarify, I am using the built in installation of impacket tools in kali linux, not through git cloning the scripts and then using python impacket_script.py

 command executed: impacket-ntlmrelayx
error:
```
Traceback (most recent call last):
  File "/usr/share/doc/python3-impacket/examples/ntlmrelayx.py", line 52, in <module>
    from impacket import version
  File "/usr/lib/python3/dist-packages/impacket/version.py", line 12, in <module>
    import pkg_resources
  File "/home/myuser/.local/lib/python3.12/site-packages/pkg_resources/__init__.py", line 90, in <module>
    import packaging.markers
  File "/home/myuser/.local/lib/python3.12/site-packages/packaging/markers.py", line 25, in <module>
    from .specifiers import InvalidSpecifier, Specifier
  File "/home/myuser/.local/lib/python3.12/site-packages/packaging/specifiers.py", line 14, in <module>
    from .utils import canonicalize_version
  File "/home/myuser/.local/lib/python3.12/site-packages/packaging/utils.py", line 9, in <module>
    from .tags import Tag, parse_tag
  File "/home/myuser/.local/lib/python3.12/site-packages/packaging/tags.py", line 7, in <module>
    import distutils.util
ModuleNotFoundError: No module named 'distutils'
```

**Comment by mubix** [comment]
Try doing `apt install python3-setuptools` that should fix it.

**Comment by sagisar1** [comment]
***Update***  

Hello, I am updating that the error was not related to impacket, it was related to a dependency conflict I had in my kali linux machine, between python packages that were installed by the system (through: _**apt install python3-setuptools**_ for example), and packages installed by pip (through _**pip install &lt;package&gt;**_).
For me, it took like three hours to resolve the issue, and I resolved it by removing the conflicting packages through:

`pip uninstall <package_name> --break-system-packages`

In my case, I had to remove in this way the packages: **_pkg-resources_**, and **_packaging_**

 
After that, I ensured that all the packages I needed, which where setuptools, packaging and pkg_resources in my case, were installed correctly through apt. In my case, only making a reinstall of python3-pkg-resources was necessary:
 
`apt install --reinstall python3-pkg-resources`

After that, I could use the native impacket binaries installed in kali linux without any errors. For anyone who got this error except me, and is using kali linux, I suggest only installing with pip through virtual environment, otherwise you will break your system, like what happened to me.

As far as I'm concerned, the issue can be closed.
